TypeCobol/Compiler/CompilationDocument.cs

```diff
@@ -541,6 +538,13 @@ public void UpdateTokensLines(System.Action onVersion = null)
                 PerfStatsForScanner.OnStopRefresh();
             }
 
+            if (scanAllDocumentLines)
```

Comment:
Why move this code here ?

Reply:
This code sends an event, it should be out of the `lock` statement.